conanfile.py

```diff
@@ -55,59 +55,21 @@ def build(self):
-            self.copy(pattern="*.h", dst="include", src=gmock_include_dir, keep_path=True)
-
-        if self.settings.os in ["Linux", "Android"] or tools.is_apple_os(self.settings.os):
-            shared_ext = "dylib" if tools.is_apple_os(self.settings.os) else "so"
-            ext = shared_ext if self.options.shared else "a"
-            self.copy("libgtest%s.%s" % (self._postfix, ext), dst="lib", src="lib")
-            if not self.options.no_main:
-                self.copy("libgtest_main%s.%s" % (self._postfix, ext), dst="lib", src="lib")
-            if self.options.build_gmock:
-                self.copy("libgmock%s.%s" % (self._postfix, ext), dst="lib", src="lib")
-                if not self.options.no_main:
-                    self.copy("libgmock_main%s.%s" % (self._postfix, ext), dst="lib", src="lib")
-        elif self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
-            self.copy(pattern="*.pdb", dst="bin", src=".", keep_path=False)
-            for ext, folder in [('lib', 'lib'), ('dll', 'bin')]:
-                self.copy("gtest%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                if not self.options.no_main:
-                    self.copy("gtest_main%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                if self.options.build_gmock:
-                    self.copy("gmock%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                    if not self.options.no_main:
-                        self.copy("gmock_main%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-        elif self.settings.os == "Windows" and self.settings.compiler == "gcc":
-            static_ext = "dll.a" if self.options.shared else 'a'
-            for ext, folder in [(static_ext, 'lib'), ('dll', 'bin')]:
-                self.copy("libgtest%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                if not self.options.no_main:
-                    self.copy("libgtest_main%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                if self.options.build_gmock:
-                    self.copy("libgmock%s.%s" % (self._postfix, ext), dst=folder, src=folder)
-                    if not self.options.no_main:
-                        self.copy("libgmock_main%s.%s" % (self._postfix, ext), dst=folder, src=folder)
+        self.copy(pattern="*.pdb", dst="bin", src=".", keep_path=False)
+        cmake = self._configure_cmake()
+        cmake.install()
 
     def package_info(self):
-        if self.options.build_gmock:
-            gmock_libs = ['gmock', 'gtest'] if self.options.no_main else ['gmock_main', 'gmock', 'gtest']
-            self.cpp_info.libs = ["{}{}".format(lib, self._postfix) for lib in gmock_libs]
-        else:
-            gtest_libs = ['gtest'] if self.options.no_main else ['gtest_main' , 'gtest']
-            self.cpp_info.libs = ["{}{}".format(lib, self._postfix) for lib in gtest_libs]
-
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
may we avoid usage of `tools.collect_libs`? it's unreliable and error prone, I've just removed it from pcre2 recipe, as it has failed in context of another recipe.

Reply:
I think it's okay for gtest, because I've checked all possible options (no_main, build_mock x settings.build_type). Both postfix and libraries are right. The only possible problem that I see is the link order, but AFAIK gtest has no problem related.